src/components/form/form.ts

```diff
@@ -0,0 +1,164 @@
+import { css, html, LitElement } from 'lit';
+import { property } from 'lit/decorators.js';
+import { EventEmitterMixin } from '../common/mixins/event-emitter';
+import { Constructor } from '../common/mixins/constructor';
+
+export interface IgcFormEventMap {
+  igcSubmit: CustomEvent<FormData>;
+  igcReset: CustomEvent;
+}
+
+/**
+ * The form is a component used to collect user input from
+ * interactive controls.
+ *
+ * @element igc-form
+ *
+ * @slot - Default slot for the form.
+ *
+ * @fires igcSubmit - Emitted when the form is submitted.
+ * @fires igcReset - Emitted when the form is reset.
+ */
+export class IgcFormComponent extends EventEmitterMixin<
+  IgcFormEventMap,
+  Constructor<LitElement>
+>(LitElement) {
+  static styles = css`
+    :host {
+      display: block;
+    }
+  `;
+
+  private _controlsWithChecked = [
+    'input',
+    'radio',
+    'igc-radio',
+    'igc-switch',
+    'igc-checkbox',
+  ];
+  private _controlsWithValue = ['input', 'igc-input', 'textarea'];
+  private _controlsThatSubmit = ['input', 'button', 'igc-button'];
+
+  /** Specifies if the form should not to be validated on submit. */
+  @property({ type: Boolean, reflect: true }) novalidate = false;
+
+  constructor() {
+    super();
+
+    this.addEventListener('click', this.handleClick);
+  }
+
+  /** Submits the form. */
+  submit(): boolean {
+    const formData = this.getFormData();
+    if (!this.novalidate && !this.reportValidity()) {
+      return false;
+    }
+
+    this.emitEvent('igcSubmit', { detail: formData });
+    return true;
+  }
+
+  /** Resets the form. */
+  reset() {
+    const formElements = this.getFormElements();
+    formElements.forEach((element) => {
+      const tagName = element.tagName.toLowerCase();
+      if (tagName === 'select') {
+        for (let i = 0; i < element.options.length; i++) {
+          const option = element.options[i];
+          option.selected = option.defaultSelected;
+        }
+      } else if (
+        (tagName === 'input' &&
+          (element.type === 'checkbox' || element.type == 'radio')) ||
+        (tagName !== 'input' && this._controlsWithChecked.includes(tagName))
+      ) {
+        element.checked = element.hasAttribute('checked');
+      } else if (tagName === 'igc-input') {
+        element.value = element.getAttribute('value');
+      } else if (this._controlsWithValue.includes(tagName)) {
+        element.value = element.defaultValue;
```

Comment:
For consistency we could use `getAttribute` for all value controls.
```suggestion
      } else if (this._controlsWithValue.includes(tagName)) {
        element.value = element.getAttribute('value');
```

Reply:
Aaand we broke the textarea... You could revert this change :)